dipy/segment/tests/test_metric.py

```diff
@@ -174,7 +174,7 @@ def dist(self, v1, v2):
             if metric.are_compatible(f1.shape, f2.shape):
                 distance = metric.dist(f1, f2)
                 if np.all(f1 == f2):
-                    assert_equal(distance, 0.)
+                    assert_almost_equal(distance, 0.)
```

Comment:
Thanks for thinking about this.   Yes, I guess it must be slight differences in the precision of dot, maybe the order or operation or something.   It's a bit surprising to me that a vector dotted with itself in float64 precision would differ by 1e-9 on two platforms, but oh well.

No need to add the special case check in the distance code, I was just curious.